src/QtImageReader.cpp

```diff
@@ -100,8 +104,15 @@ void QtImageReader::Open()
 			info.file_size = image->byteCount();
 		#endif
 		info.vcodec = "QImage";
-		info.width = image->width();
-		info.height = image->height();
+		if (!default_svg_size.isEmpty()) {
+		    // Use default SVG size (if detected)
+            info.width = default_svg_size.width();
+            info.height = default_svg_size.height();
+		} else {
+		    // Use Qt Image size as a fallback
+            info.width = image->width();
+            info.height = image->height();
+		}
```

Comment:
So we use the svg's natural size if we can for scaling later. This much I get :+1:

Reply:
Exactly, and this fallback code is for non-SVG images... which still use the QtImageReader (such as PNG or JPEG)